packages/css/src/components/file-input/file-input.scss

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+@import "../../common/text-rendering";
+
+@mixin reset-button {
+  border: 0;
+}
+
+.ams-file-input {
+  align-items: center;
+  color: var(--ams-file-input-color);
+  display: flex;
+  font-family: var(--ams-file-input-font-family);
+  font-size: var(--ams-file-input-font-size);
+  font-weight: var(--ams-file-input-font-weight);
+  justify-content: flex-start;
+  line-height: var(--ams-file-input-line-height);
+  outline-offset: var(--ams-file-input-outline-offset);
+  touch-action: manipulation;
+  width: 100%;
+
+  @include text-rendering;
+}
+
+.ams-file-input:disabled {
+  color: var(--ams-file-input-disabled-color);
+  cursor: not-allowed;
+}
+
+.ams-file-input::file-selector-button {
+  background-color: var(--ams-file-input-file-selector-button-background-color);
+  box-shadow: var(--ams-file-input-file-selector-button-box-shadow);
+  color: var(--ams-file-input-file-selector-button-color);
+  cursor: var(--ams-file-input-file-selector-button-cursor);
+  display: inline-block;
+  font-family: inherit;
+  font-size: inherit;
+  font-weight: inherit;
+  /* stylelint-disable-next-line property-disallowed-list */
+  margin-right: 2ex; // Add space between the file selector button and the file name
+  padding-block: var(--ams-file-input-file-selector-button-padding-block);
+  padding-inline: var(--ams-file-input-file-selector-button-padding-inline);
+
+  @include reset-button;
+}
+
+.ams-file-input:disabled::file-selector-button {
+  background-color: var(--ams-file-input-file-selector-button-disabled-background-color);
```

Comment:
I don't think this is necessary, the background color is always white.

Reply:
Same rules as the secondary button, I've copied them in case your input is on a different color background.

Reply:
Shouldn't the entire input get a white background color then? Same as our other inputs. It doesn't make sense to do this just for the button imo

Reply:
It is to preserve the appearance of the button and does not affect anything in a negative way.